docs/odoo/models/bus.bus.rst

```diff
@@ -1,124 +1,192 @@
+        // Creating messages
         this.openerp.session ("/send/", {message: message})
     }
 
 Create an object for widget work:
 
-.. code-block:: shell
-
-    var my_module = new MyModule.Conversation(this)
-
-For the work in the server write the following:
-
-.. code-block:: shell
+.. code-block:: js
 
-    class Controller(openerp.addons.bus.bus.Controller):
-	def _poll(self, dbname, channels, last, options):
-	if request.session.uid:
-		registry, cr, uid, context = request.registry, request.cr, request.session.uid, request.context
-		channels.append((request.db, 'module.name', request.uid))
-	return super(Controller, self)._poll(dbname, channels, last, options)
+    var my_module = new MyModule.Conversation(this);
 
-The below function will intercept form the clien the request ``/send/`` and will process this request:
+The below function will intercept form the client the request ``/send/`` and will process this request:
 
-.. code-block:: shell
+.. code-block:: py
 
     @http.route('/send/', type="json", auth="public")
-	de fmessage_send(self, message):
-	/* message processing */
-	request.env["model.name"].broadcast(message)
-	return True
+    def message_send(self, message):
+        /* message processing */
+        request.env["model.name"].broadcast(message)
+        return True
 
 ``broadcast`` function creates the notice and sends the its result (in this case, to all users except for current)
 
-.. code-block:: shell
+.. code-block:: py
 
     @api.model
     def broadcast(self, message):
-	notifications = []
-	forps in self.env['res.users'].search([('id', '!=', self.env.user.id)]):
-	notifications.append([(self._cr.dbname, 'model.name', ps.id), message])
-	self.env['bus.bus'].sendmany(notifications)
+        notifications = []
+        for ps in self.env['res.users'].search([('id', '!=', self.env.user.id)]):
+            notifications.append([(self._cr.dbname, 'model.name', ps.id), message])
+            self.env['bus.bus'].sendmany(notifications)
         return 1
 
-After sending message , function ``this.on_notification`` accepts the message. 
+**Who will get this message**
+
+After sending message , function ``this.on_notification`` accepts the message.
+
+``this.on_notification`` – is response for accepting of server messages
+Notification, which was sent from the server, includes channel and message.
+Put to the corresponding variable values from ``notification``
 
+.. code-block:: js
+
+    on_notification: function (notification) {
+        var self = this;
+        if (typeof notification[0][0] === 'string') {
+            notification = [notification]
+        }
+        for (var i = 0; i < notification.length; i++) {
+            var channel = notification[i][0];
+            var message = notification[i][1];
+            this.on_notification_do(channel, message);
+        }
+    },
+
+You should check if there are coincidences with the name of the model, from which the server's response comes.
```

Comment:
![putin-delete-it](https://cloud.githubusercontent.com/assets/186131/15927843/39b5a70a-2e5d-11e6-9674-997d58b9f67b.jpg)